ddtrace/opentelemetry/tracer_test.go

```diff
@@ -157,7 +163,15 @@ func TestShutdown(t *testing.T) {
 	assert := assert.New(t)
 	tp := NewTracerProvider()
 	otel.SetTracerProvider(tp)
+
+	testLog := new(log.RecordLogger)
+	defer log.UseLogger(testLog)()
+
 	tp.Shutdown()
+	tp.ForceFlush(5*time.Second, func(ok bool) {})
+
+	logs := testLog.Logs()
+	assert.Contains(logs[len(logs)-1], "Cannot perform (*TracerProvider).Flush since the tracer is already stopped")
```

Comment:
Nit: looks like this assertion and scenario (calling `ForceFlush` after `Shutdown`) should rather be a separate test case in `TestForceFlush`.

Reply:
I understand and appreciate the idea of covering the scenario!

Unit testing is by definition about testing the smallest testable units in the program.
In our case here, we're testing the behavior of `ForceFlush` in a `TestShutdown` unit test, which is odd especially since we have a dedicated `TestForceFlush` test. Ideally, when testing function A, the assertions shouldn't depend at all on the implementation details of another function B. In other words, changing the log string in `ForceFlush` should **not** break the unit tests of the `Shutdown()` method.

Alternatively, we can assert that `p.stopped` has been set as expected here. And in make sure we log the warning in `TestForceFlush` if `p.stopped == 1`.

Reply:
Updated 🙌